src/standalone/Snow/snow_parameterizations.jl

```diff
@@ -363,3 +370,202 @@ function energy_from_T_and_swe(S::FT, T::FT, parameters) where {FT}
+function newsnow_dens(air_temp::FT)::FT where {FT}
+    if air_temp < FT(-15.0)
+        return FT(0.05)
+    else
+        return FT(0.05) + FT(0.0017) * FT((air_temp + FT(15.0))^1.5)
+    end
+end
+
+"""
+    newsnow_dens(air_temp::FT)::FT where {FT}
+Helper function for dzdt under an `Anderson1976` density parameterization.
+A model for estimating the tempearature of newly fallen snow,
+given air temperature. Uses the model implemented in Snow17.
+"""
+function newsnow_temp(air_temp::FT)::FT where {FT}
+    return (air_temp > FT(0)) ? FT(0) : air_temp
+end
+
+"""
+    dzdt(density::Anderson1976, model::SnowModel{FT}, Y, p, t) where {FT}
+Returns the change in snow depth (rate) given the current model state and the `Anderson1976`
+density paramterization, which estimates contributions from new snowfall, as well as compaction
+effects from the existing and newly fallen snow.
+"""
+function dzdt(density::Anderson1976, model::SnowModel{FT}, Y, p) where {FT}
+    Δt_t = model.parameters.Δt / FT(3600) #in hours (when Δt in SnowParameters is ::FT instead of ::Period)
+
+    #Contribution from new snowfall: (uses parameterization of newsnow_temp, newsnow_dens)
+    air_temp = p.drivers.T .- model.parameters.earth_param_set.T_freeze
+    snowfall = abs.(p.drivers.P_snow) .* model.parameters.Δt
+    T_newsnow = newsnow_temp.(p.drivers.T)
+    ρ_precip = newsnow_dens.(air_temp) #unitless
+    ρ_newsnow =
+        compat_density.(
+            snowfall,
+            FT(0),
+            ρ_precip,
+            Δt_t,
+            T_newsnow,
+            Ref(density),
+        )
+    Δz = snowfall ./ ρ_newsnow
+
+    #Estimate resulting change from compaction effects of old snow (and the incoming snow on top):
+    W_ice = (FT(1) .- p.snow.q_l) .* Y.snow.S
+    ρ_ice = W_ice ./ Y.snow.Z #unitless
+    sliq = p.snow.q_l .* Y.snow.S
+    Tsnow = p.snow.T .- model.parameters.earth_param_set.T_freeze
+    W_use = W_ice .+ snowfall
+    parent(ρ_ice)[parent(W_ice) .<= eps(FT)] .= FT(0.1) #any nonzero value for no-snowpack to avoid NaN and return 0.0
+    ρ_est = compat_density.(W_use, sliq, ρ_ice, Δt_t, Tsnow, Ref(density))
+    dz_compaction = (W_ice ./ ρ_est) .- Y.snow.Z
+
+    return (dz_compaction .+ Δz) ./ model.parameters.Δt
+end
+
+"""
+    clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT
+A helper function which clips the tendency of Z such that
+its behavior is consistent with that of S.
+"""
+function clip_dZdt(S::FT, Z::FT, dSdt::FT, dZdt::FT, Δt::FT)::FT where {FT}
+    if (S + dSdt * Δt) < 0
+        return -Z / Δt
+    elseif (Z + dZdt * Δt) < (S + dSdt * Δt)
+        #do we need to do something about setting q_l = 1 here?
+        return (S - Z) / Δt + dSdt
+    else
+        return dZdt
+    end
+end
+
+"""
+    snow_density(density::Anderson1976, SWE::FT, z::FT, parameters::SnowParameters{FT}) where {FT}
+Returns the snow density given the current model state and the `Anderson1976`
+density paramterization.
+"""
+function snow_density(
```

Comment:
This is the same function essentially as the snow_density method for NeuralSnowDepth. I dont think we need both?